src/DataTypes/Serializations/SerializationInfo.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <Core/Block.h>
+#include <DataTypes/Serializations/ISerialization.h>
+#include <Columns/ColumnSparse.h>
+
+namespace DB
+{
+
+/** Contains information about kinds of serialization of columns.
+ *  Also contains information about content of columns,
+ *  that helps to choose kind of serialization of column.
+ *
+ *  Currently has only information about number of default rows,
+ *  that helps to choose sparse serialization.
+ *
+ *  Should be extended, when new kinds of serialization will be implemented.
+ */
+class SerializationInfo
```

Comment:
I think it may be better to make this an interface. So that complex columns like tuple or array had their own type of info, which contain info for nested columns. Kind of analogous to IDataType with nested types.

Reply:
I've implemented a a interface. Now `IDataType` doesn't require column name to get a serialization.